sirius/coding/image_encoding.py

```diff
@@ -110,21 +110,54 @@ def html_to_png(html):
     driver = None
     try:
         caps = {'acceptSslCerts': True}
-        driver = webdriver.PhantomJS(
-            'phantomjs', desired_capabilities=caps,
-            service_args=['--ignore-ssl-errors=true', '--ssl-protocol=any'])
-        driver.set_window_size(384, 5)
+        options = webdriver.ChromeOptions()
+        options.accept_untrusted_certs = True
+        options.assume_untrusted_cert_issuer = True
+
+        options.set_headless()
+
+        options.add_argument("--no-sandbox")
+        options.add_argument("--disable-impl-side-painting")
+        options.add_argument("--disable-setuid-sandbox")
+        options.add_argument("--disable-seccomp-filter-sandbox")
+        options.add_argument("--disable-breakpad")
+        options.add_argument("--disable-client-side-phishing-detection")
+        options.add_argument("--disable-cast")
+        options.add_argument("--disable-cast-streaming-hw-encoding")
+        options.add_argument("--disable-cloud-import")
+        options.add_argument("--disable-gpu")
+        options.add_argument("--disable-popup-blocking")
+        options.add_argument("--ignore-certificate-errors")
+        options.add_argument("--disable-session-crashed-bubble")
+        options.add_argument("--disable-ipv6")
+        options.add_argument("--allow-http-screen-capture")
+        options.add_argument('--force-device-scale-factor=1')
+
+        driver = webdriver.Chrome(
+            chrome_options=options,
+            desired_capabilities=caps,
+            service_args=['--ignore-ssl-errors=true', '--ssl-protocol=any']
+        )
+        driver.set_window_size(384, 8)
 
         # note that the .html suffix is required to make phantomjs
         # pick up the mime-type and render correctly.
         with tempfile.NamedTemporaryFile(suffix='.html') as f:
-            if isinstance(html, unicode):
+            if isinstance(html, str):
                 f.write(html.encode('utf8'))
             else:
                 f.write(html)
             f.flush()
             driver.get('file://' + f.name)
-            data = io.BytesIO(driver.get_screenshot_as_png())
+
+            total_height = driver.execute_script("return document.body.parentNode.scrollHeight")
+            driver.set_window_size(384, total_height)
+            screenshot = driver.get_screenshot_as_png()
+
+            # body = driver.find_element_by_tag_name('body')
```

Comment:
Don't _need_ to keep this around, but it might be more stable than resizing the viewport? It _does_ break if the height is 0 though, which is kind of undefined right now.

Reply:
Let's leave it commented there, I can imagine that it could come in handy if the window-resizing method has any issues.

Reply:
Just one thought I had - could CSS rules on the `html` element affect this? I'm guessing that `body.parentNode` is `html`... like margin/padding/border on the `html` element?

Reply:
Good question! Part of me wonders if we should apply a global CSS reset here to counter against partial/incomplete HTML layouts (say, just `<p>hello world</p>`) getting padding/margin added, but I can only imagine that going poorly when someone wants a specific layout.

The whole HTML rendering is probably fairly fragile, and it's definitely undefined right now.  I'm sure passing something simple in like `<meta name="viewport" content="width=device-width, initial-scale=2">` breaks something we care about here.

I was gonna add a bunch of tests that generate PNGs and compare against fixtures, but I'm wondering how fragile that'll be, i.e. different OS fonts will cause test failures. As long as they pass in Docker/on Circle, that's all that really matters I guess? But it'd be good to use it to define a basic spec of "enter this HTML, get this output", and keep it consistent regardless of our implementations here.

Reply:
I was wondering if just a HTML -> image height test would do the trick, although that might fall foul of the fragility problems you raise above re. fonts etc. 

HTML passed to the print key API is in fact templated, so `<p>Hello, world!</p>` would be placed within a HTML document with base styles etc. This is where the Timestamp/from header comes from.

Regardless, we're talking about very edge-casey stuff here, so don't let it hold up your progress!

Reply:
Good point about the template! I'd forgotten about that.

I added #13 as a starting point to catch any of the main regressions here. The image height test is _probably_ enough to 99% of issues, but having the image snapshots gives us a way to inspect/visualise the differences so I think has a fair bit of value.

So let's get #13 in, then get this PR green against those tests. Then let's merge it in and see how it goes.